pylint/checkers/variables.py

```diff
@@ -2254,13 +2259,25 @@ def _loopvar_name(self, node: astroid.Name) -> None:
         if not isinstance(assign, nodes.For):
             self.add_message("undefined-loop-variable", args=node.name, node=node)
             return
-        if any(
-            isinstance(
+        for else_stmt in assign.orelse:
+            if isinstance(
                 else_stmt, (nodes.Return, nodes.Raise, nodes.Break, nodes.Continue)
-            )
-            for else_stmt in assign.orelse
-        ):
-            return
+            ):
+                return
+            # TODO: 2.16: Consider using RefactoringChecker._is_function_def_never_returning
```

Comment:
This function doesn't actually infer but uses the `attrname`, which we also match `not_typing.we_do_return.NoReturn`. That isn't really safe..